### ISSUE TITLE
Lint/Debugger does not catch require "debug/start"

### ISSUE THREAD

**Issue by odinhb** [rubocop/rubocop#12721]
There is a [feature](https://github.com/ruby/debug?tab=readme-ov-file#start-by-requiring-a-library) in the [debug gem](https://rubygems.org/gems/debug) where you can require the gem and enter the debugger in a single statement. I don't think other debuggers have this so it is being missed by the `Lint/Debugger` cop.

The feature allows the following statements to start a debugger:
```rb
require 'debug/start' # Same as rdbg.
require 'debug/open' # Same as rdbg --open.
require 'debug/open_nonstop' # Same as rdbg --open --nonstop.
```

I don't think we want any of these to accidentally make it to production.

## Expected behavior

`Lint/Debugger` should complain about this code, since it is a debugger breakpoint the same way as putting `debugger` in your code is:

```rb
def a_method
  x = 2 + 2

  require "debug/start"

  x * 2
end
```

## Actual behavior

`1 file inspected, no offenses detected`

## Steps to reproduce the problem

Put `require "debug/start"` in your code. `$ rubocop `

## Attempted workaround

This has no effect:

```yml
Lint/Debugger:
  DebuggerMethods:
    requires:
      "require \"debug/start\""
```

I would assume this cop configuration does not support checking for the specific argument to require. Maybe I'm missing something?

## RuboCop version

```
$ bundle exec rubocop -V
1.56.3 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rspec 2.24.0
```

and

```
$ bundle exec rubocop -V
1.56.3 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.1) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.19.0
  - rubocop-rails 2.21.0
  - rubocop-rspec 2.24.0
```

**Comment by st0012** [comment]
`debug/open_nonstop` should be excluded from this list as it's NOT a breakpoint require. It simply activates the debugger in remote mode but doesn't stop the program. And its main usage is to be put behind a flag to enable convenient debugger-editor integration. Here's an [example](https://github.com/Shopify/ruby-lsp/blob/main/exe/ruby-lsp#L83) from the Ruby LSP. And according to this issue: https://github.com/rails/rails/pull/51225, some Rails projects also use it the same way. So please exclude it from the current list. cc @Earlopain

**Comment by koic** [comment]
@st0012 Thank you for the feedback. I've opened https://github.com/rubocop/rubocop/pull/12845 to solve it.

**Comment by st0012** [comment]
@koic Thank you very much 🙌
